xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml

```diff
@@ -617,7 +617,15 @@ $services.localization.render('core.users.unknownUser')##
   #set ($discard = $query.setLimit($rows))
   #set ($discard = $query.setOffset($start))
   #set ($discard = $query.bindValue('sort', "${sort} ${sortOrder}"))
+  #set ($discard = $query.bindValue('tie', $solrConfig.tieBreaker))
+  #set ($discard = $query.bindValue('mm', $solrConfig.minShouldMatch))
+  ## Little rudimentary but this should impart higher relevance to
+  ## documents with tags matching the search terms. Will only match single word tags at the moment
+  #set ($fieldName = $solrConfig.boostQuery[0])
+  #set ($fieldBoost = $solrConfig.boostQuery[1])
+  #set ($discard = $query.bindValue('bq', "${fieldName}:(${queryString})${fieldBoost}"))    
```

Comment:
I think the boost query makes sense if it's different than the main query. E.g. the user looks for "books" and you want to boost the books from the "science-fiction" category (even though the user didn't ask for it). In your case, since the boost query string is the same as the main query string, I think it's enough to add the field with the boost to the query fields. Don't you think?

    'queryFields': {
        'DOCUMENT': 'title^10.0 name^10.0 ... property.XWiki.TagClass.tags^1000.0 ...'

which basically means: "if the search term matches a tag then increase the score". Isn't this what you want?

Reply:
Well, I'll run you through what I would like to achieve, then it might make more sense. I wasn't convinced the `bq` parameter was the best way to go either. Hence why I was playing around with rerank.

When entering a multi term query such as `cocktail bars in New York` I'd like to check for documents with tags made up of the search terms and boost these to be the most relevant without changing the order of the results as determined by the main query. To do this I believe I need to sum a fixed boost value to the score rather than multiply the score by the boost. If more than one tag matches then multiple boost values would need summing.

In shorthand, this (sub)query would look like this:
```
q=cocktail bars in New York
tie=1.0
qf=property.XWiki.TagClass.tags^=1000
pf=property.XWiki.TagClass.tags^=1000
pf2=property.XWiki.TagClass.tags^=1000
pf3=property.XWiki.TagClass.tags^=1000
```
If tags existed for `cocktail bars` and `New York` the useful terms from the `bq` subquery that would contribute to highest scoring documents would be: `((cocktail bars)^=1000) ((New York)^=1000)`

All documents with both tags would appear as most relevant due to `1000 + 1000 + (main query document score)`. Then documents matching one tag, followed by documents with no matching tags. Note that the order of the documents within each of these "groups" (2 tags, 1 tag, no tags) is determined by the main query score.

I think I could achieve a simplified version of this type of query (without stopword removal) by composing the query using a velocity macro. But it woudn't be very elegant. Hence why I was wanting to make use of solr's natural capabilities.

The `1000` value is a little bit of a hack, but it doesn't really matter as long as it's a value larger than the highest scoring result. There might be a dynamic way of getting this value.

Reply:
I see. I don't know if you can achieve this with boost query (I haven't encountered this use case before). I guess you tried to use boost query with [local parameters](https://lucene.apache.org/solr/guide/6_6/local-parameters-in-queries.html) and it didn't work? Something like:

    $query.bindValue('bq', "{!dismax tie=1.0 qf=property.XWiki.TagClass.tags^1000 ...}$queryString")

And no, we don't have an API to serialize a Solr query to a string with local parameters like above.